writer.go

```diff
@@ -8,13 +8,35 @@ import (
 	"math"
 )
 
+// IndexVersion2 is the first recorded index version. It consists of:
+//   - NULL
+//   - backspace
+//   - ASCII character "2".
+var IndexVersion2 = []byte{0x00, 0x08, 0x32}
+
+var (
+	Version1 = 1
+	Version2 = 2
+)
+
 type rsfWriter struct {
-	writer io.Writer
-	pos    int
+	writer  io.Writer
+	version int
+	pos     int
 }
 
 func NewWriter(f io.Writer) Writer {
-	return &rsfWriter{writer: f}
+	return &rsfWriter{
+		writer:  f,
+		version: Version1,
```

Comment:
This keeps the original writer using `Version1`, which has no changes to the format. Below, I've added a new `NewWriterWithVersion` method that can be used to explicitly create a writer with a specific version, like the new `Version2`, which we'll use for CRAN RSF files.